src/vs/monaco.d.ts

```diff
@@ -1520,14 +1528,30 @@ declare namespace monaco.editor {
 		forceMoveMarkers?: boolean;
 	}
 
+	export interface IValidEditOperation {
+		/**
+		 * The range to replace. This can be empty to emulate a simple insert.
+		 */
+		range: Range;
```

Comment:
@alexdima Did you intend for this to be a `Range` rather than an `IRange`? Without tracing through the code, I assume it's extra work to relax this constraint right now?

Reply:
That is intentional. You can see how `IValidEditOperation` is always returned from the editor or passed in as a callback argument by the editor, so this is not a constrained on the editor user, it is a constrained on the editor implementation.